surch/surch.py

```diff
@@ -1,165 +1,349 @@
+        :type organization: basestring
+        :param git_user: git user name for authenticate
+        :type git_user: basestring
+        :param git_password:git user password for authenticate
+        :type git_password: basestring
+        :param local_path: this path contain the repos clone
+        :type local_path: basestring
+        :param verbose: user verbose mode
+        :type verbose: bool
+        """
+        self.search_list = search_list
+        self.organization = organization
+        self.git_user = git_user
+        self.git_password = git_password
+        self.ignore_repo = skipped_repo
+        self.local_path = local_path
+        lgr.setLevel(logging.DEBUG if verbose else logging.INFO)
+
+    @classmethod
+    def from_config_file(cls, config_file, verbose=False):
+        """Define vars from "config.yaml" file"""
         with open(config_file, 'r') as config:
             conf_vars = yaml.load(config.read())
-        self.search_list = conf_vars.get('search_list')
-        self.ignore_repo = conf_vars.get('skipped_repo')
-        self.organization = conf_vars.get('organization')
-        self.git_cred = conf_vars.get('git_user') + ':' + conf_vars.get('git_password')
-        self.git_user = conf_vars.get('git_user')
-        self.git_password = conf_vars.get('git_password')
-        self.local_path = conf_vars.get('local_path')
-
-    def get_github_repo_list(self, user, password, organization, repo_per_page=100000):
-        ''' api request for git url to clone'''
+        conf_vars.setdefault('verbose', verbose)
+        return cls(**conf_vars)
+
+    def get_github_repo_list(self, git_user, git_password, organization, repo_kind='public',
+                             url_kind='clone_url', repo_per_page=100):
+        """
+        This method get from git hub the git url list for clonnig
+        :param git_user: git user name for authenticate
+        :type git_user: basestring
+        :param git_password:git user password for authenticate
+        :type git_password: basestring
+        :param organization: organization name
+        :type organization: basestring
+        :param repo_kind: repository kind (all, private, public, fork) default:'public'
+        :type repo_kind: basestring
+        :param url_kind: url kind (git_url, ssh_url, clone_url, svn_url) default:'clone_url'
+        :type url_kind: basestring
+        :param repo_per_page: this for getting the MAX information in 1 page default:100
+        :type repo_per_page: int
+        :return:
+        """
         github_repo_urls = []
-        data = requests.get(GET_GIT_REPO_DETAILS_API_URL.format(organization, repo_per_page), auth=(user, password))
-        data = data.json()
-        for repo_data in data:
-            github_repo_urls.append(repo_data['clone_url'])
+        data = requests.get(GET_ORG_DETAILS.format(organization), auth=(git_user, git_password))
+
+        repo_num = data.json()['{0}_repos'.format(repo_kind)]
```

Comment:
handle cases where a user requests repo types which do not exist  in the organization.